vespa/application.py

```diff
@@ -1517,9 +1543,17 @@ def _open_httpx_client(self):
         limits = httpx.Limits(
             max_keepalive_connections=self.connections,
             max_connections=self.connections,
-            keepalive_expiry=10,  # This should NOT exceed the keepalive_timeout on the Server, otherwise we will get ConnectionTerminated errors.
+            keepalive_expiry=self.keepalive_expiry,  # This should NOT exceed the keepalive_timeout on the Server, otherwise we will get ConnectionTerminated errors.
         )
-        timeout = httpx.Timeout(pool=5, connect=5, read=5, write=5)
+        if self.total_timeout:
+            timeout = httpx.Timeout(self.total_timeout)
```

Comment:
If I understand the following code correctly this is not really total_timeout but more like default value for different timeouts:
https://github.com/encode/httpx/blob/1bf1fc0ea8417c8e539914aeafdb48f15591ba75/httpx/_config.py#L246
Perhaps we can let users pass Timeout object so we don't have to deal with the logic of how to set timeouts? This does break abstraction layer so kind of a tradeoff.

Reply:
Yes, good point. Tried to clarify in docstring for `total_timeout`-parameter. 
Did it this way to ensure backwards compatibility (by not changing parameter name). 
Good suggestion to let user pass Timeout. 
My suggestion is then to deprecate `total_timeout` parameter (issue `DeprecationWarning`,  and introduce `timeout: httpx.Timeout = httpx.Timeout(5)`. 
Does that sound reasonable?

Reply:
Sounds good.